sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -859,6 +859,31 @@ object SparkSession extends Logging {
       this
     }
 
+    /**
+     * Sets a config option. Options set using this method are automatically propagated to
+     * both `SparkConf` and SparkSession's own configuration.
+     *
+     * @since 3.4.0
+     */
+    def config(map: Map[String, Any]): Builder = synchronized {
+      map.foreach {
+        kv: (String, Any) => {
+          options += kv._1 -> kv._2.toString
```

Comment:
### choose
1. type check -> config(String, type) method
2. foreach -> options += key -> value

---

I chose 2.
because it is simple code.